icefoganalytics/travel-authorization
# Add Meals and Incidentals Table to Expense Tab

Fixes https://github.com/icefoganalytics/travel-authorization/issues/139

Relates to:
- [Analysis, Expense Pages, 2023-12-19](https://docs.google.com/document/d/1cWEA-0w9ro1TyajfnffkYaKL42yYnA89i9gdQysht_A/edit?pli=1#heading=h.zc2nwn7nmjlu)

# Context

**Is your feature request related to a problem? Please describe.**
Need a simple table, generated from travel expense prefill, so that a user can view there current travel authorization expenses related to meals and incidentals.
Add note below Expenses table saying that the user doesn't need to add meals and incidentals there.

**Describe the solution you'd like**
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/a0857ee1-7771-4c85-915d-3068e3b941d9)

**Additional context**
See `api/src/services/expenses/prefill-service.ts` for logic on generating expenses from estimates. This service will need some additions.
- `web/src/modules/travel-authorizations/pages/EditMyTravelAuthorizationExpensePage.vue` will need to have the table logic restricted to exclude meals and incidientals

# Implementation

Upgrade to Vue 2.7 so that I can use <script setup> style.
Add "use" pattern for expenses so I can maintain independent per-query state for an object, but still track loading state.
Tweak prefill service so that it also generates meals and accommodation expenses.

# Screenshots

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/a436d18f-5d32-4a38-b6ab-4dae9e9674f1)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "My Travel Requests" page from the top nav drop down.
5. Create a new travel authorization via the "+ Travel Authorization" button.
6. Fill in the various sections, and then click "Generate Estimate".
7. Submit the form to a supervisor, that isn't you.
8. Change the status of the travel authorization to approved via the database console.
9. Refresh the page to get the Expense tab to unlock.
10. Go to Expense tab and click the "Prefill" expense button.
11. This now prefills the editable expenses and the non-editable meals and incidentals expenses.
12. Check that the generated data matches the appropriate estimates.
